src/modules/expression/unop/neg.rs

```diff
@@ -51,6 +57,20 @@ impl TranslateModule for Neg {
     }
 }
 
+impl Neg {
+    pub fn get_integer_value(&self) -> Option<isize> {
+        self.expr.get_integer_value().map(isize::neg)
+    }
```

Comment:
Good suggestion. I'll give it a go.

Reply:
Hmm. On second thoughts, I'm not sure that's such a good idea, because it returns a `Result` not an `Option`.  Yes I know you can convert the one to the other, but returning a `Result` implies that it's an error condition if it fails to parse, rather than expected behaviour.

Reply:
Yep. I agree with @hdwalters. On paper this seems like a good idea but in reality it doesn't really fit into our architecture.

Reply:
Ok, that's resolved then.